synapse/lib/cell.py

```diff
@@ -2587,6 +2583,95 @@ async def _initCellBoot(self):
         async with s_telepath.loadTeleCell(self.dirn):
             await self._bootCellMirror(provconf)
 
+    @classmethod
+    async def _initBootRestore(cls, dirn):
+
+        env = 'SYN_RESTORE_HTTPS_URL'
+        rurl = os.getenv(env, None)
+
+        if rurl is None:
+            return
+
+        dirn = s_common.gendir(dirn)
+
+        # restore.done - Allow an existing URL to be left in the configuration
+        # for a service without issues.
+        doneiden = None
+
+        donepath = s_common.genpath(dirn, 'restore.done')
+        if os.path.isfile(donepath):
+            with s_common.genfile(donepath) as fd:
+                doneiden = fd.read().decode().strip()
+
+        rurliden = s_common.guid(rurl)
+
+        if doneiden == rurliden:
+            return
+
+        # If there is a mismatch vs restore.done, we hulk smash in the restored contents.
+        # There are no seatbelts here to check for existing files / etc.
+
+        rnfo = s_urlhelp.chopurl(rurl)
+
+        clean_url = s_urlhelp.sanitizeUrl(s_urlhelp.zipurl(rnfo))
+        logger.info(f'Restoring {cls.getCellType()} from {env}={clean_url}')
+
+        # Setup get args
+        rnfo_query = rnfo.pop('query', {})
+        # DISCUSS - SHOULD WE POP THIS OUT OF THE URL ENTIRELY IF PRESENT?
+        syn_ssl_verify = rnfo_query.get('syn_ssl_verify')
```

Comment:
I'd def be concerned here about scope mismatch and mixed logic/data concerns. It would probably be preferable to use an optional `+no-verify` in the protocol setting that can be cleanly re-written. Or perhaps use `httpx://` to denote TLS without validity checking.

Reply:
could definitely just use the schema as a hint there. we need to support environments where a backup may be coming from a potentially untrusted https endpoint.